internal/build/job/maker.go

```diff
@@ -105,9 +111,13 @@ func (m *maker) MakeJobTemplate(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.B
 		kanikoImageTag = buildConfig.KanikoParams.Tag
 	}
 
+	dockefileData, err := m.getDockerfileContents(ctx, buildConfig, mod.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get dockerfile contents: %v", err)
+	}
 	specTemplate := v1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Annotations: map[string]string{"Dockerfile": buildConfig.Dockerfile},
+			Annotations: map[string]string{"Dockerfile": dockefileData},
```

Comment:
I would rather mount the ConfigMap into the pod.

Reply:
i will check if kaniko supports that option

Reply:
+1

Reply:
Fixed. Please review the changes:
1) new Volume definition\
2) new hash calculation
3) changed build config pickup implementation